### ISSUE TITLE
Search Page Improvements for Screen Reader support

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#11204]
- This ticket contains improvements for [Headings and Labels Success Criterion](https://github.com/SAP/spartacus/issues/11192)  ([WCAG 2.1 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html))
- master ticket: #11192

## Labels improvements 

- [x] No aria-label on view switcher
- [x] No aria-label on Relevance dropbox (part of ng-select refactoring, separate issue)
- [x] Remove product button need better description - "Remove Product from Cart"
- [x] Close modal button needs better description - "Close Modal"
- [ ] No aria-label category facet buttons (patters in not accessible via keyboard. more investigation is needed. reference more details in the table of issues) open a separate issue (Separate issue created: #12898)

## Update / Fix things that break
- [x] Update Markup
- [x] Update SCSS
- [x] Fix / Update unit tests
- [x] Fix / Update e2e tests

![Screen Shot 2021-02-16 at 1 28 02 PM](https://user-images.githubusercontent.com/15215400/108532312-5667b080-729d-11eb-91b8-171b6debe376.png)

add to cart modal 
![Screen Shot 2021-02-16 at 1 32 54 PM](https://user-images.githubusercontent.com/15215400/108532337-5b2c6480-729d-11eb-918d-6b26a6270d94.png)
